### PR TITLE
Notebook updates: Getting Started & Explore Warming

### PR DESCRIPTION
**Pull request by vicford**
This PR updates the getting_started and explore_warming notebooks with our most recent updates to climakitae. 

Specifically:
- Markdown reference to climakitae library standardized across both notebooks (I also checked other notebooks, and this is what is consistently done).
- getting_started now has app.view and a customizable version included in place of the existing visualize code cells
- explore_warming now has export functionality included

### CODE REVIEW

File: explore_warming.ipynb
Comment (nicolejkeeney) [inline explore_warming.ipynb:319]: I think beth removed the functionality for this in app.export. If I'm not mistaken, we now assume the user provides the function the data in the form that they want to export. So, we should delete this line.

File: explore_warming.ipynb
Comment (nicolejkeeney) [inline explore_warming.ipynb:322]: Similarly, I'm not sure if you can do this anymore. I would recommend consulting Beth.

File: getting_started.ipynb
Comment (nicolejkeeney) [inline getting_started.ipynb:18]: The wording here might be better written as: 
"This cell imports the python library climakitae, our AE toolkit for climate data analysis."

File: getting_started.ipynb
Comment (nicolejkeeney) [inline getting_started.ipynb:39]: I think the description in parentheses here is a bit redundant. Would recommend deleting.

File: getting_started.ipynb
Comment (nicolejkeeney) [inline getting_started.ipynb:149]: I would recommend moving `data_to_use.compute()` into the previous section "Step 2: Retrieve Data". Because it is more about data retrieval, not visualization. By calling compute(), we load the data into memory. I also don't like how the dataset is renamed from `data_to_use` to just `data`; I think it's a bit confusing, because it's the same data, just loaded into memory. In step two, I would recommend adding a cell and a description: 


"Next, load the data into memory. This allows the previous steps to run faster"
`data_to_use = data_to_use.compute()`

**Review by nicolejkeeney — CHANGES_REQUESTED** [review]
See comments.

File: explore_warming.ipynb
Comment (nicolejkeeney) [reply explore_warming.ipynb:322]: This will need to be modified in getting_started.ipynb as well.

File: explore_warming.ipynb
Comment (nicolejkeeney) [reply explore_warming.ipynb:319]: This will need to be modified in getting_started.ipynb as well.

**Comment by elehmer** [comment]
A couple of issues I noticed with export:

1) The GeoTiff didn't have any projection info when loaded into ArcGIS Pro.
2) The CSV had some values for `degrees warmer` field are blank (not sure if this is ok or not)

Otherwise it runs flawlessly!

**Comment by elehmer** [comment]
> A couple of issues I noticed with export in `explore_warming` notebook:
> 
> 1. The GeoTiff didn't have any projection info when loaded into ArcGIS Pro.
> 2. The CSV had some values for `degrees warmer` field are blank (not sure if this is ok or not)
> 
> Otherwise it runs flawlessly!

Looks like #2 is just the NoData cells now that I look at other CSV outputs.

**Comment by vicford** [comment]
> * The GeoTiff didn't have any projection info when loaded into ArcGIS Pro.
> * The CSV had some values for `degrees warmer` field are blank (not sure if this is ok or not)

@bethmints - can you check on these? These might be potential export errors, not notebook errors

**Comment by elehmer** [comment]
> > > * The GeoTiff didn't have any projection info when loaded into ArcGIS Pro.
> > > * The CSV had some values for `degrees warmer` field are blank (not sure if this is ok or not)
> > 
> > 
> > @bethmints - can you check on these? These might be potential export errors, not notebook errors
> 
> I'm looking into this in more depth as well to see what is happening...

Ahh it's because we reproject the plot data to geographic:

```
"gridded = plots.plot.pcolormesh('lon','lat', col='simulation',\n",
    "                     transform=ccrs.PlateCarree(),subplot_kws={'projection': ccrs.Orthographic(-100,40)},\n",
    "                         figsize=[15,5],cmap='Reds')\n",
```

We don't have a provision for dealing with this in export

**Comment by elehmer** [comment]
> > > > * The GeoTiff didn't have any projection info when loaded into ArcGIS Pro.
> > > > * The CSV had some values for `degrees warmer` field are blank (not sure if this is ok or not)
> > > 
> > > 
> > > @bethmints - can you check on these? These might be potential export errors, not notebook errors
> > 
> > 
> > I'm looking into this in more depth as well to see what is happening...
> 
> Ahh it's because we reproject the plot data to geographic:
> 
> ```
> "gridded = plots.plot.pcolormesh('lon','lat', col='simulation',\n",
>     "                     transform=ccrs.PlateCarree(),subplot_kws={'projection': ccrs.Orthographic(-100,40)},\n",
>     "                         figsize=[15,5],cmap='Reds')\n",
> ```
> 
> We don't have a provision for dealing with this in export

Oh wait, we don't export `gridded`...now I'm confused...

**Comment by vicford** [comment]
> Oh wait, we don't export `gridded`...now I'm confused...sorry about that. The GeoTiff data is in Lam. Con. but just doesn't have the projection info...I'll dig at the code some more.

Beth's suggestion was to remove the geotiff option for export all together for now, and to just provide to_csv and to_netcdf options for the AMY dataframe

**Comment by elehmer** [comment]
> > Oh wait, we don't export `gridded`...now I'm confused...sorry about that. The GeoTiff data is in Lam. Con. but just doesn't have the projection info...I'll dig at the code some more.
> 
> Beth's suggestion was to remove the geotiff option for export all together for now, and to just provide to_csv and to_netcdf options for the AMY dataframe

Yeah, good idea. We need to examine what is going on during data transformations. Some how we lost `grid_mapping` again and it was replaced with ```grid_mapping_name : lambert_conformal_conic```. My guess would be it happened in `get_anomaly_data`. I will dig into more.

**Comment by vicford** [comment]
> Beth's suggestion was to remove the geotiff option for export all together for now, and to just provide to_csv and to_netcdf options for the AMY dataframe

ACK SORRY - this is specifically in reference to **only** the AMY notebook! This is what happens when I do multiple notebook PRs at once.... This **should** be addressed for _getting_started_ and _explore_warming_!

**Comment by elehmer** [comment]
> 

Oh! Sorry I misread that lol. Yeah, ok I'll try and figure this out.

**Comment by elehmer** [comment]
> > Beth's suggestion was to remove the geotiff option for export all together for now, and to just provide to_csv and to_netcdf options for the AMY dataframe
> 
> ACK SORRY - this is specifically in reference to **only** the AMY notebook! This is what happens when I do multiple notebook PRs at once.... This **should** be addressed for _getting_started_ and _explore_warming_!

Oh I misread your comment! I see, ok, I will try and figure out where it goes off the rails

**Comment by elehmer** [comment]
> > > Beth's suggestion was to remove the geotiff option for export all together for now, and to just provide to_csv and to_netcdf options for the AMY dataframe
> > 
> > 
> > ACK SORRY - this is specifically in reference to **only** the AMY notebook! This is what happens when I do multiple notebook PRs at once.... This **should** be addressed for _getting_started_ and _explore_warming_!
> 
> Oh I misread your comment! I see, ok, I will try and figure out where it goes off the rails

Yes, so the data does lose all it's attributes between step 7 and 8 where `get_anomoly_data` is called. If you look at Line 120 of `warming_levels.py` you will see that you create a new xarray DataSet and then add data to it, but the attributes get left behind from the input data. Fixing that should fix down stream code such as export

**Comment by bethmints** [comment]
hi all! yes, I was going to say that I copy all attributes from the data array, so if the geotiff is missing some metadata, it's because it got lost upstream.

**Review by elehmer — APPROVED** [review]

**Comment by vicford** [comment]
Just waiting on @bethmints confirmation of export markdown before merging

**Comment by elehmer** [comment]
All looks good except I found a bug while exporting to CSV. If you have certain characters in the file name they will be striped out when creating the metadata file for CSVs. See Issue #102 in climakitae.

**Comment by bethmints** [comment]
> All looks good except I found a bug while exporting to CSV. If you have certain characters in the file name they will be striped out when creating the metadata file for CSVs. See Issue #102 in climakitae.

hm, I thought I fixed that in my last PR. will check, thank you.

**Comment by bethmints** [comment]
the export text looks appropriate to me! we could potentially do more hand-holding -- we'll see how users do, I suppose.

**Review by bethmints — APPROVED** [review]

**Pull request merged**
